File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = -1
    s = list(s)
    stack = []
    
    for i in range(len(s)) :
        if len(stack) != 0 :
            if s[i] == stack[-1] :
                stack.pop()
            else :
                stack.append(s[i])
        else :
            stack.append(s[i])
        
    return 1 if len(stack) == 0 else 0

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.js
function solution(s)
{
    arr = s.split('');
    stack = [];
    const reg = /(.)\1+/;
    
    for(let i = 0 ; i < arr.length; i++){
        if(arr[i] === stack.at(-1)){
            stack.pop();
        }
        else{
            stack.push(arr[i]);
        }
    }
    
    return reg.test(s) === true ? (stack.length === 0 ? 1 : 0 ) : 0 ;
}